File: erpnext_france/erpnext_france/report/bank_transaction_categories/bank_transaction_categories.js
// Copyright (c) 2022, Dokos SAS and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Bank Transaction Categories"] = {
	"filters": [
		{
			fieldname: "company",
			label: __("Company"),
			fieldtype: "Link",
			options: "Company",
			default: frappe.defaults.get_user_default("Company"),
			reqd: 1
		},
		{
			fieldname: "period",
			label: __("Period"),
			fieldtype: "Select",
			options: [
				{ "value": "this month", "label": __("This Month") },
				{ "value": "last month", "label": __("Last Month") },
				{ "value": "this quarter", "label": __("This quarter") },
				{ "value": "last quarter", "label": __("Last quarter") },
				{ "value": "this year", "label": __("This Year") },
				{ "value": "last year", "label": __("Last Year") },
			],
			default: "1 year",
			reqd: 1
		},
		{
			fieldname: "transaction_type",
			label: __("Transaction Type"),
			fieldtype: "Select",
			options: [
				{
					"label": __("Revenue"),
					"value": "credit"
				},
				{
					"label": __("Expense"),
					"value": "debit"
				}
			],
			reqd: 1
		},
	]
};

File: erpnext_france/erpnext_france/report/bank_transaction_categories/bank_transaction_categories.py
# Copyright (c) 2022, Dokos SAS and contributors
# For license information, please see license.txt

import frappe
from frappe import _


def execute(filters=None):
	columns, data = get_columns(filters), get_data(filters)
	chart_data = get_chart_data(data)
	return columns, data, None, chart_data


def get_columns(filters=None):
	return [
		{"fieldtype": "Data", "fieldname": "category", "label": _("Category"), "width": 250},
		{"fieldtype": "Currency", "fieldname": "amount", "label": _("Amount"), "width": 300},
	]


def get_data(filters=None):
	return frappe.get_list(
		"Bank Transaction",
		filters={
			"date": ("Timespan", filters["period"]),
			"company": filters["company"],
			filters.get("transaction_type"): (">", 0.0),
		},
		fields=["category", f'sum({filters.get("transaction_type")}) as amount'],
		group_by="category",
	)


def get_chart_data(data):
	return {
		"data": {
			"labels": [d.category or _("Without category") for d in data],
			"datasets": [{"name": "Amount", "values": [d.amount for d in data]}],
		},
		"type": "pie",
		"height": 500,
		"fieldtype": "Currency",
	}
